src/GitVersionCore.Tests/CommitDateTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using GitVersion;
+using GitVersion.VersionFilters;
+using NUnit.Framework;
+
+namespace GitVersionCore.Tests
+{
+    [TestFixture]
+    public class CommitDateTests
+    {
+        public static IEnumerable<TestCaseData> CommitDateFormatTestCases
+        {
+            get
+            {
+                var date = new DateTime(2017, 10, 6);
+
+                var testCasesFormats = new[]
+                {
+                    "yyyy-MM-dd",
+                    "dd.MM.yyyy",
+                    "yyyyMMdd",
+                    "yyyy-MM"
+                };
+
+                foreach (var format in testCasesFormats)
+                    yield return new TestCaseData(date, format, date.ToString(format));
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(CommitDateFormatTestCases))]
```

Comment:
Since the test case formats are constant strings, please add them as regular `[TestCase]` attributes instead of the more complicated `TestCaseData`.